### DISCUSSION TITLE
Oqtane Template Structure

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#1571] [Q&A]
Hi,

Is there any detailed article / docs for Oqtane template structure?

How do I customize top navbar, content area, etc.,?

How do I create new containers or sections in the template?

How do I change "Oqtane Admin Panel" model popup into page?

How do I implement different template or themes for each Sites and one admin template in Oqtane?

**Suggestions:**
Better to separate razor UI components to Razor Class Library project in the Oqtane framework. Which will have all template or themes resources file and razor UI components.

Thanks & Regards!

**Comment by AndreasDecke** [comment]
**I would also be interested.**

I don't think the template (skin) strategy from DNN is good. Often the template for a normal page (home etc.) is much more complex than the template for an admin page. The manual setting (of a theme) for each individual admin page is very annoying.

_Greetings and thanks_

**Reply by sbwalker** [reply]
@AndreasDecke Oqtane uses concepts that are similar to DNN but uses its own approach when it comes to theming. For example, Oqtane does not have a "Portal Theme" and an "Admin Theme" that is defined at the Site level - it only has a "Default Theme". So my suggestion in Oqtane is that you would do an assessment of your site architecture and you would set the default theme for the site to be the theme which is expected to be used in the majority of cases ( which might be the admin theme if your site only has a few content pages ). Then you would manually override the theme at the page level on only those pages which require a different theme. This keeps the manual effort to a minimum.

**Reply by AndreasDecke** [reply]
Thanks for the answer. I had already thought about what the best practice would be. And I also came to this approach.

**Comment by leigh-pointer** [comment]
@Sharafudeen I would suggest looking at the Client project where you will find two themes.  These give a very clear outline on how it hangs together.  [https://github.com/oqtane/oqtane.framework/tree/dev/Oqtane.Client/Themes](https://github.com/oqtane/oqtane.framework/tree/dev/Oqtane.Client/Themes)

From there you can login to the framework, select Themes from the Management Panel and select Themes.  You will be able then to create theme from the theme template.  Navigate to where the theme was created and open the solution.

**Reply by Sharafudeen** [reply]
@leigh-pointer What I'm looking for is to customize in new theme completely without changing default Oqtane's themes (Oqtane & Blazor themes). To do this.. I've created new theme which is created from Oqtane theme management. 

I need some clarifications or suggestions to do the followings,

1. Add Syncfusion Top Navbar - How do I get datasource (pages) in new theme project from Oqtane framework? 
2. Top Navbar contains all pages, logo, user profile, login, language switcher and control panel menus. How do I get all its data or api in new theme project?

**Reply by sbwalker** [reply]
@Sharafudeen if you want to create a custom menu component based on Syncfusion then I would suggest you look at the default menu components in oqtane.framework\Oqtane.Client\Themes\Controls\Theme. You will see that the list of pages is available in the PageState property which is part of ThemeBase.

**Reply by Sharafudeen** [reply]
@sbwalker Thanks for your answer. I have done without changing top navbar since I'm planning to implement [https://github.com/oqtane/oqtane.theme.bootswatch](https://github.com/oqtane/oqtane.theme.bootswatch) themes in my project. Bootswatch themes are great look & feel and Bootstrap 5 is the feature!
